lib/rules/selector-max-type/__tests__/index.js

```diff
@@ -549,3 +549,82 @@ testRule(rule, {
     }
   ]
 });
+
+testRule(rule, {
+  ruleName,
+  config: [0, { ignore: ["next-sibling"] }],
+  skipBasicChecks: true,
+  syntax: "scss",
+
+  accept: [
+    {
+      code: ".foo + a {}"
+    },
+    {
+      code: ".foo, .bar + a {}"
+    },
+    {
+      code: ".foo + a + b {}"
+    },
+    {
+      code: ".foo { .bar + a {} }"
+    },
+    {
+      code: ".foo, .bar + baz {}"
+    },
+    {
+      code:
+        ".foo + bar + foo + baz + bar, .foo + baz, .baz + .foo { .baz + foo + bar + foo { .foo + foo {} } }, .lol + .lol {}"
+    }
+  ],
+
+  reject: [
+    {
+      code: "foo {}",
+      message: messages.expected("foo", 0),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "bar baz {}",
+      message: messages.expected("bar baz", 0),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "bar + baz {}",
+      message: messages.expected("bar + baz", 0),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "foo + bar + baz {}",
+      message: messages.expected("foo + bar + baz", 0),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "foo, .bar + baz {}",
+      message: messages.expected("foo", 0),
+      line: 1,
+      column: 1
+    },
+    {
+      code: ".foo + bar, baz {}",
+      message: messages.expected("baz", 0),
+      line: 1,
+      column: 13
+    },
+    {
+      code: ".foo + .bar { baz {} }",
+      message: messages.expected(".foo + .bar baz", 0),
+      line: 1,
+      column: 15
+    },
+    {
+      code: ".foo + .bar { .baz + .foo { foo + bar {} } }",
```

Comment:
Let's use standard syntax here by adding the nesting selector:

```suggestion
      code: ".foo + .bar { & .baz + .foo { & foo + bar {} } }",
```